Assets/Scripts/R&D Tree/ResearchNode.cs

```diff
@@ -0,0 +1,134 @@
+using UnityEngine;
+using UnityEngine.Events;
+using UnityEngine.UI;
+
+public class ResearchNode : MonoBehaviour
+{
+    [SerializeField] private Button _button;
+    [SerializeField] private Slider _researchBar;
+
+    private float _timer;
+
+    public UnityEvent ResearchDoneEvent = new UnityEvent();
+
+    private ResearchNodeSetting _nodeSetting;
+
+    public SkillState CurrentSkillState;
+
+    private void OnEnable()
+    {
+        SetUIForStates();
+    }
+
+    private void Start()
```

Comment:
Maybe also a good plan to combine nodes and nodesettings since they do pretty much the same

Reply:
**Edit** Maybe not, since the content and rewards of the should not be directly linked. Maybe even create an event or callback we can set outside of the R&D tree codebase to set the rewards or logic that should run when the node is researched